### PR TITLE
Upgrade Node to v20

### PR DESCRIPTION
**Pull request by aparnajyothi-y**
**Description:**
Changes added for the node-version upgrade from 16 to 20 for actions/setup-java

**Related issue:**
Feature request

**Check list:**
- [x] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: 'Setup Java JDK'
+nnname: 'Setup Java JDK'
```
Comment (dmitry-shibanov) [inline action.yml:1]: Could you please revert this changes.

File: action.yml [hunk 2]
```diff
@@ -78,6 +78,6 @@ outputs:
   cache-hit:
     description: 'A boolean value to indicate an exact match was found for the primary key'
 runs:
-  using: 'node16'
+  using: 'node21'
```
Comment (dmitry-shibanov) [inline action.yml:81]: Could you please change it to node20. If you used the `npm ci` command with node 21 on local machine you need to change it node20 and regenerate dependencies with it.

File: package.json [hunk 3]
```diff
@@ -29,28 +29,35 @@
...
-    "ts-jest": "^27.0.5",
-    "typescript": "^4.2.3"
+    "ts-jest": "^29.1.1",
+    "typescript": "^5.3.2"
+  },
+  "bugs": {
+    "url": "https://github.com/actions/setup-java/issues"
+  },
+  "homepage": "https://github.com/actions/setup-java#readme",
+  "directories": {
```
Comment (dmitry-shibanov) [inline package.json:60]: You can remove directories part.

File: action.yml (hunk 1 above)
Comment (aparnajyothi-y) [reply action.yml]: @dmitry-shibanov  reverted the changes. Thank you.

File: action.yml (hunk 2 above)
Comment (aparnajyothi-y) [reply action.yml]: @dmitry-shibanov  Updated node version to 2o and build sucessful in my local.

**Comment by aparnajyothi-y** [comment]
@MaksimZhukov  @dmitry-shibanov I have resolved the conflicts. Please review. Thank you.

File: .github/workflows/check-dist.yml [hunk 4]
```diff
@@ -15,3 +15,5 @@ jobs:
   call-check-dist:
     name: Check dist/
     uses: actions/reusable-workflows/.github/workflows/check-dist.yml@main
+with:
```
Comment (MaksimZhukov) [inline .github/workflows/check-dist.yml:18]:
```suggestion
    with:
```

**Review by MaksimZhukov — APPROVED** [review]

File: .github/workflows/check-dist.yml (hunk 4 above)
Comment (aparnajyothi-y) [reply .github/workflows/check-dist.yml]: Updated the format and removed the conflicts. Please review.

**Review by MaksimZhukov — APPROVED** [review]

**Review by IvanZosimov — APPROVED** [review]
